index.bs

```diff
@@ -1739,6 +1782,219 @@ contexts that have active documents; navigation can also cause contexts to
 become inaccessible but not yet get discarded because bfcache.
 </div>
 
+## Log ## {#module-log}
+
+The <dfn export for=modules>log</dfn> module contains functionality and events
+related to logging.
+
+### Definition ### {#module-log-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+
+Command = (GetCommand)
+</pre>
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+
+LogEvent = (
+  LogEntryAddedEvent
+)
+</pre>
+
+### Types ### {#module-log-types}
+
+#### log.LogEntry #### {#types-log-logentry}
+
+<pre class="cddl local-cddl">
+
+LogLevel = "debug" / "info" / "warning" / "error"
+
+LogEntry = {
+  type: text,
+  level: LogLevel,
+  text: text / null,
+  timestamp: int,
+  ?stackTrace: [*StackFrame],
+  extra: { *text => any }
+}
+
+ConsoleLogExtra = {
+  type: text,
+  realm: Realm,
+  args: [*RemoteValue],
+}
+
+JavascriptLogExtra = {
+  lineNumber: int,
+  columnNumber: int,
+  url: text
+}
```

Comment:
Is it somehow different from the `?stackTrace: [*StackFrame]` in the `LogEntry`?

Reply:
It definitely feels like this might duplicate the stack information. We can remove it.